seqr/views/utils/variant_utils.py

```diff
@@ -131,15 +132,22 @@ def _add_locus_lists(projects, genes, add_list_detail=False, user=None):
 
 
 def _get_rna_seq_outliers(gene_ids, family_guids):
-    data_by_individual_gene = defaultdict(lambda: {'outliers': {}})
-
-    outlier_data = get_json_for_rna_seq_outliers(
-        RnaSeqOutlier.objects.filter(
-            gene_id__in=gene_ids, p_adjust__lt=RnaSeqOutlier.SIGNIFICANCE_THRESHOLD, sample__individual__family__guid__in=family_guids),
-        nested_fields=[{'fields': ('sample', 'individual', 'guid'), 'key': 'individualGuid'},]
-    )
-    for data in outlier_data:
-        data_by_individual_gene[data.pop('individualGuid')]['outliers'][data['geneId']] = data
+    data_by_individual_gene = defaultdict(lambda: {'outliers': {}, 'spliceOutliers': {}})
+
+    for outlier, outlier_cls, max_sign in [('outliers', RnaSeqOutlier, None),
+                                           ('spliceOutliers', RnaSeqSpliceOutlier, MAX_SIGNIFICANT_OUTLIER_NUM)]:
+        outlier_data = get_json_for_rna_seq_outliers(
+            outlier_cls.objects.filter(
+                gene_id__in=gene_ids,
+                sample__individual__family__guid__in=family_guids,
+                sample__is_active=True,
+            ).order_by('p_value'),
+            nested_fields=[{'fields': ('sample', 'individual', 'guid'), 'key': 'individualGuid'},
+                           {'fields': ('sample', 'tissue_type'), 'key': 'tissueType'}],
```

Comment:
Isn't that a lot of duplicated data though? Like I would expect there to only be a couple sample models returned, but this way you would be returning tissue type hundereds of times

Reply:
If we return `sampleGuid`, the `sampleGuid` will be duplicated hundreds of times, just like the `tissueType`. Returning `tissueType` saves steps to get the `tissueType` from the sample data.

Reply:
But if we return `sampleGuid` only `sampleGuid` will be duplicated, whereas now we are duplicating `tissueType` and `individualGuid`

Reply:
Also, in the UI getting the possible tissue types will be much faster iterating over a couple samples than hundreds of rnaseq models

Reply:
The `rnaSeqData` in the response is organized on a per-individual basis. So the `individualGuid` won't be duplicated. If we return the tissue type, no iteration is required in the UI. It just uses the tissue type in the data. The only thing is that we need to map the single or two letters tissue types (e.g., 'M', 'F', etc.) to displayable strings. But the costs of lookup the mapping table are smaller than looking up the sample table.

Reply:
If you have a nested field for `individualGuid`, as you do above, it will be added for every object

We define basically all the enumerations in constants.js and do the mapping on the client side, and return single letter codes on the server side, so that is probably the right approach

Reply:
We need the nested field `individualGuid`. The field will be popped and used as a key in the response. This differs from the individual API, which has the `individualGuid` before fetching the outlier models and no need for a nested `individualGuid` field.

Reply:
I think we are talking past each other. I think there are some high level questions that need to be aligned on before figuring out the best approach here. 

1. Why do we need `tissueType` at all here? In the new UI you are showing tissue type is not shown
2. Why are we customizing `get_json_for_rna_seq_outliers` so much here? In general, we do not want our json helpers to be overly customizable, we want to be really consistent everywhere they are used. Ideally we would have the same json for these everywhere, but in the event that we can't we need to be explicit about what is being added. In particular, a) why would the `max_significant_num_per_tissue` be different here than in other places, that should be a property of the model class? And b) why are the nested fields required here different? If the nested fields do need to be different `get_json_for_rna_seq_outliers` should accept a flag like `add_individual_guid_field` or something specific like that

Reply:
1. I should have added a `tissue type` column according to https://github.com/broadinstitute/seqr/issues/3200#:~:text=hover%20over%20the%20tag%20should%20show%20info%20about%20the%20splice%20junction%20outlier%20(chrom/start/end/strand%2C%20z%2Dscore%2C%20pvalue%2C%20maybe%20tissue%20and%20read%20count%20also) "hover over the tag should show info about the splice junction outlier (chrom/start/end/strand, z-score, pvalue, `maybe tissue` and read count also)"

2. We can avoid overly customizing `get_json_for_rna_seq_outliers` by creating a new function `get_json_for_splice_outliers`. a) there is only one value for `max_significant_num_per_tissue` when it is not `None`. If we separate a function for splice junction outliers, it is a constant. b) the variant search API gets the outliers per family, while the individual API gets them per individual. This difference makes the code for the two APIs different, but they share the same `get_json...` function. And the variant search API needs the nested individual GUID field, and the individual API doesn't.

Reply:
The ticket says "maybe tissue" and we will be leaving tissue out for MVP, so you do not need to add it.

I still don't why the functionality here needs to be refactored so much. There is already a function to get the data for the volcano plot page, which by definition will contain all the info needed for the the junction table. Why can't we reuse that ?